### DISCUSSION TITLE
High concurrent  visiting neo4j database ,appears IO ERROR " io driver has terminated". How to solve it?

### DISCUSSION THREAD

**Discussion by pengyizeng** [tokio-rs/tokio#4420] [Q&A]
Recently, I try "neo4rs"," Deadpool-bolt", "b88-bolt" to create connection pool with neo4j. My program appears

"error:UnexpectedMessage("unexpected response for RUN: Err(IOError { detail: \"IO driver has terminated\" })")"driver has terminated" 

when many query appears in a short time.  How to solve it ?

I have google the error , but I did not found any useful information.

**Comment by Darksonn** [comment]
This error means that the Tokio runtime has been shut down, but something is still trying to use an IO resource from the runtime.

**Reply by carllerche** [reply]
It sounds like maybe this is part of a termination process that isn't graceful? If so, you can look at mini-redis to see how shutdown is done there. E.g. [here](https://github.com/tokio-rs/mini-redis/blob/master/src/server.rs#L333) and see the rest of the file.
